crates/runtime/src/datafusion.rs

```diff
@@ -818,9 +826,35 @@ impl DataFusion {
                     table_name: dataset_name.to_string(),
                 })?;
         }
+
         Ok(())
     }
 
+    pub async fn get_accelerated_table_provider(
```

Comment:
this method doesn't return a &AcceleratedTable due to lifetime issue that `table` can't be used in returned reference.

Let me know if any better way to deal with this.